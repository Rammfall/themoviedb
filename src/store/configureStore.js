import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
