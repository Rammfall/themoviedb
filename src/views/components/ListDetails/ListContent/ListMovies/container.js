import React from 'react'
import { connect } from 'react-redux'
import { compose } from 'ramda'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import MoviesList from 'Views/components/MoviesList'
import { getListDetailsMoviesSelector } from 'Store/concepts/listDetails/selectors'

const ListMovies = ({ total, movies }) => (
  <MoviesList
    movies={movies}
    total={total}
  />
)

ListMovies.propTypes = {
  total: PropTypes.number.isRequired,
  movies: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      original_title: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
}

const mapStateToProps = (state, { match: { params: { id } } }) => {
  const { movies, total } = getListDetailsMoviesSelector(state, id)

  return {
    total,
    movies
  }
}

export { ListMovies as ListMoviesContainer }
export default compose(
  withRouter,
  connect(mapStateToProps)
)(ListMovies)
