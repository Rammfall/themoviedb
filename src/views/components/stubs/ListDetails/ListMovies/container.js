import React from 'react'
import { connect } from 'react-redux'
import { compose } from 'ramda'
import { withRouter } from 'react-router-dom'

import { getListMoviesSelector } from 'Store/concepts/movies/selectors'
import MoviesList from 'Views/components/MoviesList'

const ListMovies = ({ moviesList: { total, movies } }) => (
  <MoviesList
    movies={movies}
    total={total}
  />
)

const mapStateToProps = (state, { match: { params: { id } } }) => ({
  moviesList: getListMoviesSelector(state, id)
})

export default compose(
  connect(mapStateToProps),
  withRouter
)(ListMovies)
