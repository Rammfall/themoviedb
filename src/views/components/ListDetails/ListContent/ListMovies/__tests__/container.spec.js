import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import diveTo from 'TestUtils/diveToEnzyme'

import ListMovies, { ListMoviesContainer } from '../container'

jest.mock('react-router-dom', () => ({
  withRouter: component => component
}))

jest.mock('Store/concepts/listDetails/selectors', () => ({
  getListDetailsMoviesSelector: jest.fn(() => ({
    total: 1,
    movies: [{
      original_title: 'test',
      id: 1
    }]
  }))
}))

describe('ListMovies', () => {
  it('matches snapshot', () => {
    const store = configureStore()()
    const props = {
      match: {
        params: {
          id: 1
        }
      }
    }

    const wrapper = shallow(
      <ListMovies
        store={store}
        {...props}
      />
    )
    const container = diveTo(wrapper, ListMoviesContainer)

    expect(container).toMatchSnapshot()
  })
})
