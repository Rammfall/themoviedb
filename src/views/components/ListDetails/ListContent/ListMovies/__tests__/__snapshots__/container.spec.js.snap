// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMovies matches snapshot 1`] = `
<MoviesList
  actions={null}
  movies={
    Array [
      Object {
        "id": 1,
        "original_title": "test",
      },
    ]
  }
  total={1}
/>
`;
